in_toto/gpg/common.py

```diff
@@ -334,11 +337,61 @@ def _assign_certified_key_info(bundle):
             "by '{}'.".format(signature["keyid"]))
         continue
 
-      # TODO: If the signature is valid, extract relevant information from
-      # its "info" field and assign to master key (e.g. expiration) here
-      # NOTE: Beware of conflicting information. There might be multiple User
-      # IDs per primary key and multiple signatures per User ID. See RFC4880
-      # 5.2.3.19. and last paragraph of 5.2.3.3. for more info about ambiguity.
+      # If the signature is valid, we try to extract subpackets relevant to
+      # the primary key, i.e. expiration time.
+      # NOTE: There might be multiple User IDs per primary key and multiple
+      # certificates per User ID. RFC4880 5.2.3.19. and last paragraph of
+      # 5.2.3.3. provides some suggestions about ambiguity, but delegates the
+      # responsibility to the implementer.
+
+      # Ambiguity resolution scheme:
+      # We take the key expiration time from the most recent certificate, i.e.
+      # the certificate with the highest signature creation time. Additionally,
+      # we prioritize certificates with primary user id flag set True. Note
+      # that, if the ultimately prioritized certificate does not have a key
+      # expiration time subpacket, we don't assign one, even if there were
+      # certificates of lower priority carrying that subpacket.
+      tmp_validity_period = \
+          signature["info"]["subpackets"].get(KEY_EXPIRATION_SUBPACKET)
+
+      # No key expiration time, go to next certificate
+      if tmp_validity_period == None:
+        continue
+
+      tmp_sig_creation_time = \
+          signature["info"]["subpackets"].get(SIG_CREATION_SUBPACKET)
+
+      # No signature creation time, no way to resolve potential ambiguities,
+      # go to next certificate
+      # TODO: As per the spec this cannot not happen, should we blow up?
```

Comment:
I do think we'd like to blow up in this case, or log a warning at least.

Reply:
Good point! I slightly refactored the parsing in https://github.com/in-toto/in-toto/pull/266/commits/f071c95f954a9beb7398470f4a4391b11d61b8f8. Please see the commit message for more details.

We still don't blow up, but we log the problem, and with the refactor, blowing up will be easier  in the future  (also see the updated `TODO` comment).